include/OpenSpaceToolkit/Astrodynamics/Estimator/TLESolver.hpp

```diff
@@ -0,0 +1,210 @@
+    );
+
+    /// @brief Access solver
+    ///
+    /// @return Least squares solver
+    const LeastSquaresSolver& accessSolver() const;
+
+    /// @brief Access satellite number
+    ///
+    /// @return Satellite number
+    const Integer& accessSatelliteNumber() const;
+
+    /// @brief Access international designator
+    ///
+    /// @return International designator
+    const String& accessInternationalDesignator() const;
+
+    /// @brief Access revolution number
+    ///
+    /// @return Revolution number
+    const Integer& accessRevolutionNumber() const;
+
+    /// @brief Access whether to fit with B*
+    ///
+    /// @return Whether to fit with B*
+    const bool& accessFitWithBStar() const;
+
+    /// @brief Access estimation frame
+    ///
+    /// @return Estimation frame
+    const Shared<const Frame>& accessEstimationFrame() const;
+
+    /// @brief Access default B* value
+    ///
+    /// @return Default B* value
+    const Real& accessDefaultBStar() const;
+
+    /// @brief Access first derivative of mean motion divided by 2
+    ///
+    /// @return First derivative of mean motion divided by 2
+    const Real& accessFirstDerivativeMeanMotionDividedBy2() const;
+
+    /// @brief Access second derivative of mean motion divided by 6
+    ///
+    /// @return Second derivative of mean motion divided by 6
+    const Real& accessSecondDerivativeMeanMotionDividedBy6() const;
+
+    /// @brief Access ephemeris type
+    ///
+    /// @return Ephemeris type
+    const Integer& accessEphemerisType() const;
+
+    /// @brief Access element set number
+    ///
+    /// @return Element set number
+    const Integer& accessElementSetNumber() const;
+
+    /// @brief Access TLE state builder
+    ///
+    /// @return TLE state builder
+    const StateBuilder& accessTLEStateBuilder() const;
+
+    /// @brief Estimate TLE from observations
+    ///
+    /// @param anInitialGuess Initial guess (TLE, State+BStar pair, or State)
+    /// @param anObservationArray Observations
+    /// @param anInitialGuessSigmas Initial guess sigmas
+    /// @param anObservationSigmas Reference state sigmas
+    Analysis estimateTLE(
+        const std::variant<TLE, Pair<State, Real>, State>& anInitialGuess,
+        const Array<State>& anObservationArray,
+        const std::unordered_map<CoordinateSubset, VectorXd>& anInitialGuessSigmas = DEFAULT_INITIAL_GUESS_SIGMAS,
+        const std::unordered_map<CoordinateSubset, VectorXd>& anObservationSigmas = DEFAULT_OBSERVATION_SIGMAS
+    ) const;
+
+   private:
+    LeastSquaresSolver solver_;
+    Integer satelliteNumber_;
+    String internationalDesignator_;
+    Integer revolutionNumber_;
+    bool fitWithBStar_;
+    Shared<const Frame> estimationFrameSPtr_;
+
+    mutable Real defaultBStar_;
+    Real firstDerivativeMeanMotionDividedBy2_;
+    Real secondDerivativeMeanMotionDividedBy6_;
+    Integer ephemerisType_;
+    Integer elementSetNumber_;
+    StateBuilder tleStateBuilder_;
```

Comment:
This feels like a lot of memeber variables, maybe we can group them better with an intermediate object or something?

Reply:
We could make a TLE Solver Configuration object but it'll kinda be the same thing, lots of members. Let me noodle with it.